.github/workflows/codeql-analysis.yml

```diff
@@ -7,17 +7,19 @@ name: "CodeQL"
 
 on:
 # Disable for push and pull_request until https://github.com/keycloak/keycloak/issues/10203 is resolved
-#  push:
-#    branches: [main]
-#  pull_request:
+  push:
+    branches: [main]
+  pull_request:
```

Comment:
I might be missing something, but given:

> For security reasons, GitHub won't give permissions to a pull-request to read `GITHUB_TOKEN` and write security events. So, it is expected that all the checks in this PR for CodeQL will fail.

after this PR is merged to `main` are we going to have it working on PRs or should we only do the scan on `main`?

Reply:
@andreaTP yes, please check the working PoC added to the description. Also, I suggest experimenting it by copying those files to your fork.